### DISCUSSION TITLE
Example for a simple custom highlighter

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#2190] [Q&A] [answered]
Currently try to wrap my head around adding a own custom highlighter (In this case for inlineHilite), but I struggle to really understand how it would work here...

My goal is to have a highlighting similar to HTML, where the content within `<text>` tags is changed... Just extended to also include `[]` and `{}` variants (While not changing special characters like `|`).

Quick mockup I made using dev tools:
![grafik](https://github.com/facelessuser/pymdown-extensions/assets/11576465/04ecfab5-b8a2-4d86-9f44-21e7d48e6cc3)

Could you give a small example that could explain this to me a bit better? Would apreciate it a lot as I have a hard time finding the right page explaining this further...

**Comment by facelessuser** [comment]
You are asking about a custom Pygments highlighter?

**Reply by Andre601** [reply]
Yes. I ask about a basic example of highlighting code with a custom highlighter.

Have some troubles understanding it... I know I define a method that in my case probs should return HTML, but I can't quite figure out where I would put this file and if it needs anything else...
Is there a doc explaining this?

**Reply by facelessuser** [reply, marked as answer]
If are wanting highlighting, you should write a Pygments lexer, install it, and then it will be available, no custom inline needed. I have a couple custom lexers I've written here: https://github.com/facelessuser/pymdown-lexers/tree/master/pymdown_lexers.

**Reply by Andre601** [reply]
Okay. Need to look at it a bit...
I assume I add it to pygment itself? Or would I do that within the configuration somewhere?

**Reply by facelessuser** [reply]
Pygments requires that the extension is registered under the appropriate entry point. https://github.com/facelessuser/pymdown-lexers/blob/master/setup.py#L5

Now, can custom lexers be used without registering them under the entry point? Maybe? But I haven't dug too deep into that. If there is a way to dynamically load them without registering them under an entry point, I'd love to know how to do that, I just haven't looked into it.

**Reply by Andre601** [reply]
Okay... I looked into pygment itself to see how they do it within their HTML lexer and... my head hurts now 😅 

I never was really good at decoding regex here, so I'll pass on this custom highlighting stuff for now (The HTML highlighting aproach I use right now works good enough for me).
Thanks tho for the detailed info. Apreciate it a lot.

**Reply by facelessuser** [reply]
You could do it without a lexer. I have one where we do mathjax wrapping: https://github.com/facelessuser/pymdown-extensions/blob/main/pymdownx/arithmatex.py#L127. You could replace that code with some other approach if you like.

**Reply by Andre601** [reply]
Still don't quite understand it tho... But again, no need here for you to waste any more of your time on me.

**Reply by facelessuser** [reply]
Okay. Just for anyone coming here. Basically you can do custom inline blocks just like you can do custom fences. You can just return an etree element.

**Reply by Andre601** [reply]
I've worked a bit on a own pygments Lexer to use... The only question I still have open right now is, how would I implement it in pygments to be used?

I've looked through their docs, but have a bit of a hard time understanding it. Tho from what I get would I need to register this through Python?
If so, would a MkDocs hook work here? Since I would use this for MkDocs.

I understand if this question is out of scope for this repository and can forward it to the pygments one if wanted.

For the sake of clarity is here the pygments discussion I started when working on the custom lexer: https://github.com/pygments/pygments/discussions/2539

**Reply by facelessuser** [reply]
As far as I'm aware, you create a package that you can install via pip. That's what I do: https://github.com/facelessuser/pymdown-lexers. If there is a dynamic way that doesn't require creating an installable package, then I'd love to know, but currently, I just create my own package and add as many lexers as I want.

**Reply by Andre601** [reply]
Okay. I found a StackOverflow answer that works for me... Tho it comes with deprecation warnings I would like to avoid.
Tho, the file setup seems to be the same as the one on your repository down to the setup.py's content (Tho mine is a lot more stripped down).

I've asked in the pygment discussion for alternative solutions. Perhaps someone there has an answer that could also help you. :D

**Reply by facelessuser** [reply]
My setup should be modernized with a `project.toml` when I get around to it. Then I'd kill off the `setup.py`, but as far as I know, that's the way to go. But if you learn something different, I'm more than happy to hear.
